kornia/augmentation/container/base.py

```diff
@@ -1,18 +1,19 @@
 from collections import OrderedDict
-from typing import Any, Iterator, List, NamedTuple, Optional, Tuple, Union
+from typing import Any, Dict, Iterator, List, NamedTuple, Optional, Tuple, Union
 
 import torch
 import torch.nn as nn
 
 from kornia.augmentation import MixAugmentationBaseV2
 from kornia.augmentation.base import _AugmentationBase
+from kornia.core import Module, Tensor
 
 __all__ = ["SequentialBase", "ParamItem"]
 
 
 class ParamItem(NamedTuple):
     name: str
-    data: Optional[Union[dict, list]]
+    data: Optional[Union[Dict[str, Tensor], List['ParamItem']]]
```

Comment:
> The `test-cpu` CI failing seems to be related to #2027, but the typing CI is failing because of `--cobertura-xml-report` flag for mypy. Should I try to debug the problem, or should we remove the typing coverage report ([because of this issue](https://github.com/kornia/kornia/pull/2024#discussion_r1030960546))?

What is causing the `seg fault` is this change `List['ParamItem']` -- I will undo this change, and check later if is a mypy bug

Reply:
https://github.com/python/mypy/issues/14200